mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/uwazi/widgets/UwaziRelationShipWidget.java

```diff
@@ -0,0 +1,183 @@
+package rs.readahead.washington.mobile.views.fragment.uwazi.widgets;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageButton;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+
+import com.google.gson.Gson;
+import com.google.gson.internal.LinkedTreeMap;
+import com.google.gson.reflect.TypeToken;
+
+import org.hzontal.shared_ui.buttons.PanelToggleButton;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import rs.readahead.washington.mobile.R;
+import rs.readahead.washington.mobile.presentation.uwazi.UwaziRelationShipEntity;
+import rs.readahead.washington.mobile.views.collect.widgets.QuestionWidget;
+import rs.readahead.washington.mobile.views.custom.CollectRelationShipPreviewView;
+import rs.readahead.washington.mobile.views.fragment.uwazi.entry.UwaziEntryPrompt;
+
+
+@SuppressLint("ViewConstructor")
+public class UwaziRelationShipWidget extends UwaziQuestionWidget {
+    private ArrayList<UwaziRelationShipEntity> relationShipEntities = new ArrayList<>();
+
+    Context context;
+    Button addEntities;
+    ImageButton clearButton;
+    ViewGroup entitiesPanel;
+    PanelToggleButton entitiesToggle;
+    TextView numberOfEntities;
+    View infoEntitiesPanel;
+    private final List<UwaziRelationShipEntity> items;
+
+    public UwaziRelationShipWidget(Context context, @NonNull UwaziEntryPrompt formEntryPrompt, boolean isPdf) {
+        super(context, formEntryPrompt);
+        items = formEntryPrompt.getEntities();
+
+        this.context = context;
+        LinearLayout linearLayout = new LinearLayout(getContext());
+        linearLayout.setOrientation(LinearLayout.VERTICAL);
+
+
+        addImageWidgetViews(linearLayout);
+        addAnswerView(linearLayout);
+        setHelpTextView(getContext().getString(R.string.Uwazi_RelationShipWidget_Select_Entities));
+        clearAnswer();
+        if (items != null) showPreviewFromDraft();
+
+    }
+
+
+    @Override
+    public ArrayList<UwaziRelationShipEntity> getAnswer() {
+        return relationShipEntities;
+
+    }
+
+    @Override
+    public void setFocus(Context context) {
+
+    }
+
+    @Override
+    public String setBinaryData(@NonNull Object data) {
+        ArrayList<UwaziRelationShipEntity> result = new ArrayList<>();
+
```

Comment:
Use List instead of ArrayList for more general type safety :  

`List<UwaziRelationShipEntity> result = new ArrayList<>();`